src/node_contextify.cc

```diff
@@ -1224,15 +1224,18 @@ static void MeasureMemory(const FunctionCallbackInfo<Value>& args) {
                                                           args[1].As<Object>());
     CHECK_NOT_NULL(sandbox);
     context = sandbox->context();
-    if (context.IsEmpty()) {  // Not yet fully initilaized
+    if (context.IsEmpty()) {  // Not yet fully initialized
       return;
     }
   }
-  v8::Local<v8::Promise> promise;
-  if (!isolate->MeasureMemory(context, static_cast<v8::MeasureMemoryMode>(mode))
-           .ToLocal(&promise)) {
-    return;
-  }
+  Local<Promise::Resolver> resolver =
+      v8::Promise::Resolver::New(context).ToLocalChecked();
```

Comment:
```suggestion
  Local<Promise::Resolver> resolver;
  if (!Promise::Resolver::New(context).ToLocal(&resolver)) return;
```

Reply:
Is this over protective ? Our codebase use this initialization quite a lot if you search `Promise::Resolver::New(`

Reply:
I only see one use of `Promise::Resolver::New()` with `.ToLocalChecked()`. (And yes, that one should be changed, imo.)

Either way, please don’t use `.ToLocalChecked()` when it’s not absolutely guaranteed that the operation will succeed. And if the operation was guaranteed to succeed, the method wouldn’t return a `MaybeLocal`.

Plus, maybe more importantly, handling exceptions should be the *standard* pattern in our source code. We use `.ToLocalChecked()` and friends far too much, and when people write new code, they tend to imitate or copy those patterns.

Reply:
Thanks for the explaination.